SingularityService/src/test/java/com/hubspot/singularity/data/ValidatorTest.java

```diff
@@ -80,4 +81,35 @@ public void itForbidsHealthCheckStartupDelaysLongerThanKillWait() {
         .contains("Health check startup delay");
   }
 
+  @Test
+  public void itForbidsZeroPortsWithHealthCheckConfiguration() {
+    HealthcheckOptions healthCheck = new HealthcheckOptionsBuilder("/").build();
+    Resources resources = new Resources(0.125, 256, 0);
+    SingularityDeploy deploy = SingularityDeploy
+        .newBuilder("1234567", "1234567")
+        .setHealthcheck(Optional.of(healthCheck))
+        .setResources(Optional.of(resources))
+        .build();
+    SingularityRequest request = new SingularityRequestBuilder("1234567", RequestType.SERVICE).build();
+
+    WebApplicationException exn = (WebApplicationException) catchThrowable(() -> validator.checkDeploy(request, deploy));
+    assertThat((String) exn.getResponse().getEntity())
+        .contains("Either an explicit port number, or port resources and port");
+  }
+
+  @Test
+  public void itIgnoresZeroPortsWithHealthCheckConfigurationButSkipped() {
+    HealthcheckOptions healthCheck = new HealthcheckOptionsBuilder("/").build();
+    Resources resources = new Resources(0.125, 256, 0);
+    SingularityDeploy deploy = SingularityDeploy
+        .newBuilder("1234567", "1234567")
+        .setHealthcheck(Optional.of(healthCheck))
+        .setResources(Optional.of(resources))
+        .setSkipHealthchecksOnDeploy(Optional.of(Boolean.TRUE))
+        .build();
+    SingularityRequest request = new SingularityRequestBuilder("1234567", RequestType.SERVICE).build();
+    WebApplicationException exn = (WebApplicationException) catchThrowable(() -> validator.checkDeploy(request, deploy));
+    assertThat((String) exn.getResponse().getEntity())
+        .doesNotContain("Either an explicit port number, or port resources and port");
```

Comment:
This is a rather loose sort of assertion since the test does error, but lower down in `checkDeploy`. Unfortunately that method is complex and fairly poorly tested as is. I did my best to capture a reasonable-ish approximation.